ios/engine/KMEI/KeymanEngine/Classes/Manager.swift

```diff
@@ -1074,183 +617,71 @@ UIGestureRecognizerDelegate {
     } else {
       let deviceParam = (UIDevice.current.userInterfaceIdiom == .phone) ? "iphone" : "ipad"
       let encodedUrl = jsonUrl.absoluteString.addingPercentEncoding(withAllowedCharacters: .urlPathAllowed)!
-      url = URL(string: "\(apiRemoteURL)\(encodedUrl)&device=\(deviceParam)")!
+      url = URL(string: "\(apiRemoteURL)\(encodedUrl)&dateformat=seconds&device=\(deviceParam)")!
     }
 
     guard let data = try? Data(contentsOf: url) else {
       let error = NSError(domain: "Keyman", code: 0,
                           userInfo: [NSLocalizedDescriptionKey: "Failed to fetch JSON file"])
-      downloadFailed(forKeyboard: kbInfo, error: error)
+      downloadFailed(forKeyboards: [], error: error)
       return
     }
 
-    let jsonObj: Any
+    let decoder = JSONDecoder()
+    decoder.dateDecodingStrategy = .secondsSince1970
     do {
-      jsonObj = try JSONSerialization.jsonObject(with: data, options: .mutableContainers)
+      let keyboard = try decoder.decode(Keyboard.self, from: data)
+      return downloadKeyboard(keyboard)
     } catch {
-      downloadFailed(forKeyboard: kbInfo, error: error as NSError)
-      return
-    }
-
-    guard let jsonDict = jsonObj as? [AnyHashable: Any] else {
-      let error = NSError(domain: "Keyman", code: 0,
-                          userInfo: [NSLocalizedDescriptionKey: "The keyboard could not be installed"])
-      downloadFailed(forKeyboard: kbInfo, error: error)
+      downloadFailed(forKeyboards: [], error: error as NSError)
```

Comment:
Is there a reason we're changing the type of error thrown here?

Reply:
No the `as NSError` is unnecessary but was probably inserted before `downloadFailed()` was changed to take `Error` instead of `NSError`.

Reply:
Actually it is needed since `downloadFailed()` requires an `NSError`. This is changed in #389 I believe.